frontend/modules/tree/tree.selectors.ts

```diff
@@ -16,14 +16,279 @@
+	selectTreeProccessing, selectDataRevision,
+	(treeProcessingData) => treeProcessingData.visibilityMap
+);
+
+export const selectInvisibleNodesIds = createSelector(
+	selectTreeProccessing, selectDataRevision,
+	(treeProcessingData) => treeProcessingData.invisibleNodesIds || []
+);
+
+export const selectNodesIndexesMap = createSelector(
+	selectTreeProccessing, selectDataRevision,
+	(treeProcessingData) => treeProcessingData.nodesIndexesMap
+);
+
+export const selectNodesBySharedIdsMap = createSelector(
+	selectTreeProccessing, selectDataRevision,
+	(treeProcessingData) => treeProcessingData.nodesBySharedIdsMap
+);
+
+export const selectMeshesByModelId = createSelector(
+	selectTreeProccessing, selectDataRevision,
+	(treeProcessingData) => treeProcessingData.meshesByModelId
+);
+
+export const selectSelectedFilters = createSelector(
+	selectComponentState, (state) => state.selectedFilters
+);
+
+export const selectFilteredNodesList = createSelector(
+	selectTreeNodesList, selectSelectedFilters, (nodes, selectedFilters) => {
+		if (!selectedFilters.length) {
+			return nodes;
+		}
+		return searchByFilters(nodes, selectedFilters, true);
+	}
+);
+
+export const selectSearchEnabled = createSelector(
+	selectComponentState, (state) => state.searchEnabled
+);
+
+export const selectIfcSpacesHidden = createSelector(
+	selectComponentState, (state) => state.ifcSpacesHidden
+);
+
+export const selectDefaultHiddenNodesIds = createSelector(
+	selectDefaultVisibilityMap, (nodesVisibilityMap) => {
+		return keys(pickBy(nodesVisibilityMap, (selectionState) => {
+			return selectionState === VISIBILITY_STATES.INVISIBLE;
+		}));
+	}
+);
+
+export const selectExpandedNodesMap = createSelector(
+	selectTreeDomain, (state) => state.expandedNodesMap
+);
+
+export const selectVisibleTreeNodesList = createSelector(
+	[
+		selectFilteredNodesList, selectSelectedFilters,
+		selectExpandedNodesMap, selectSearchEnabled
+	],
+	(treeNodesList, selectedFilters, expandedNodesMap, searchEnabled) => {
+		const visibleNodes = [];
+		const indexesByRootParentIds = {};
+
+		for (let index = 0; index < treeNodesList.length; index++) {
+			const treeNode = { ...treeNodesList[index] };
+
+			const isSearchActive = searchEnabled && selectedFilters.length;
+			const isFirstLevel = treeNode.level === 1;
+			const isSecondLevel = treeNode.level === 2;
+
+			treeNode.isSearchResult = isSearchActive && !treeNode.isFederation && !treeNode.isModel;
+			treeNode.isRegularNode = !isSearchActive && (isFirstLevel || isSecondLevel || expandedNodesMap[treeNode.parentId]);
+			if (treeNode.isSearchResult || treeNode.isRegularNode) {
+				visibleNodes.push(treeNode);
+
+				if (!treeNode.rootParentId) {
+					indexesByRootParentIds[treeNode._id] = visibleNodes.length - 1;
+				} else {
+					treeNode.rootParentIndex = indexesByRootParentIds[treeNode.rootParentId];
+				}
+			}
+
+		}
+
+		return visibleNodes;
+	}
+);
+
+export const getSelectNodesByIds = (nodesIds) => createSelector(
```

Comment:
tbh, we used that naming to emphasize that the function returns selector. :P

Reply:
I kind of figured it may be that. But it's kind of confusing due to select has a special meaning in tree.

I think it may be better to make it as a prefix before the get? I. E. selectGetNodesById

Reply:
👍

Reply:
✅